src/borg/helpers.py

```diff
@@ -1068,7 +1068,7 @@ class Location:
     ssh_re = re.compile(r"""
         (?P<proto>ssh)://                                   # ssh://
         """ + optional_user_re + r"""                       # user@  (optional)
-        (?P<host>[^:/]+)(?::(?P<port>\d+))?                 # host or host:port
+        (?P<host>([^:/]+|\[[^/]+\]))(?::(?P<port>\d+))?     # host or host:port or [ipv6] or [ipv6]:port
```

Comment:
"[" + "anything except /" + "]" is a rather unspecific match for a v6 address.
could this be more specific easily to avoid it going wrong somehow? like hex digits and colons?
where does the "except /" come from? I would have expected it matching everything until it finds "]"?